aptos-move/e2e-move-tests/tests/code_publishing.rs

```diff
@@ -302,3 +302,54 @@ fn code_publishing_using_resource_account() {
     );
     assert_success!(result);
 }
+
+#[derive(Serialize, Deserialize)]
+struct ModuleData {
+    resource_signer_cap: AccountAddress,
+}
+
+#[test]
+fn init_module_for_resource_account() {
+    let mut h = MoveHarness::new();
+    let acc = h.new_account_at(AccountAddress::from_hex_literal("0xcafe").unwrap());
+
+    let mut pack = PackageBuilder::new("Package1").with_policy(UpgradePolicy::arbitrary());
+    pack.add_source("m", "module 0x0b6beee9bc1ad3177403a04efeefb1901c12b7b575ac5124c0205efc0dd2e32a::m { public fun f() {} }");
+    let pack_dir = pack.write_to_temp().unwrap();
+    let package = framework::BuiltPackage::build(
+        pack_dir.path().to_owned(),
+        framework::BuildOptions::default(),
+    )
+        .expect("building package must succeed");
+
+    let code = package.extract_code();
+    let metadata = package
+        .extract_metadata()
+        .expect("extracting package metadata must succeed");
+    let bcs_metadata = bcs::to_bytes(&metadata).expect("PackageMetadata has BCS");
```

Comment:
actually this is to bytes, this cannot fail unless there's something like an out of memory issue.

So "must serialize" or "expected to serialize"